### PR TITLE
♻️ Small refactor stage checks

### PR DESCRIPTION
**Pull request by znatty22**

### CODE REVIEW

File: kf_lib_data_ingest/etl/ingest_pipeline.py [hunk 1]
```diff
@@ -171,41 +167,16 @@ def run(self):
...
-                # Collapse stage subtypes to their bases for storage
-                # (e.g. GuidedTransformStage -> TransformStage)
-                stage_type = type(stage)
-                while stage_type.__base__ != IngestStage:
-                    stage_type = stage_type.__base__
-
-                self.stage_outputs[stage_type] = output
-
+                # Standard stage output validation
                 if concept_discovery_dict:
```
Comment (fiendish) [inline kf_lib_data_ingest/etl/ingest_pipeline.py]: Maybe don't bother to return it if the return isn't going to be used. check stage.concept_discovery_dict instead?

Comment (fiendish) [inline kf_lib_data_ingest/etl/ingest_pipeline.py]: Maybe don't bother to return it if the return isn't going to be used. check stage.concept_discovery_dict instead?

Comment (znatty22) [reply kf_lib_data_ingest/etl/ingest_pipeline.py]: 👍

File: kf_lib_data_ingest/common/stage.py [hunk 2]
```diff
@@ -153,6 +166,16 @@ def run(self, *args, **kwargs):
         # Write output of stage to disk
         self.write_output(output)
 
+        # Write data for accounting to disk
         concept_discovery_dict = self._postrun_concept_discovery(output)
-
-        return output, concept_discovery_dict
+        self.concept_discovery_dict = concept_discovery_dict
```
Comment (fiendish) [inline kf_lib_data_ingest/common/stage.py]: need this `concept_discovery_dict`? why not `self.concept_discovery_dict = ...`

Comment (znatty22) [reply kf_lib_data_ingest/common/stage.py]: 👌

**Pull request merged**